core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -1118,8 +1118,13 @@ class SocketServerTest extends JUnitSuite {
         "Connections not registered", waitTimeMs = 15000)
       assertEquals(Set.empty, errors)
       testableSelector.waitForOperations(SelectorOperation.Register, numConnections)
+
+      // In each iteration, SocketServer processes atmost connectionQueueSize (1 in this test)
+      // new connections and then does poll() to process data from existing connections. So for
+      // 5 connections, we expect 5 iterations. Since we stop when the 5th connection is processed,
+      // we can safely check that there were atleast 4 polls prior to the 5th connection.
       val pollCount = testableSelector.operationCounts(SelectorOperation.Poll)
-      assertTrue(s"Connections created too quickly: $pollCount", pollCount >= numConnections)
+      assertTrue(s"Connections created too quickly: $pollCount", pollCount >= numConnections - 1)
```

Comment:
Isn't it easier to just say >?

Reply:
@ijuma Thanks for the review. Did you mean `pollCount > numConnections - 2`? I was thinking `-1` makes it more obvious that the last one was subtracted.

Reply:
My bad, I misread. Your code looks good.